driver-core/src/main/java/com/datastax/driver/core/policies/ErrorAwarePolicy.java

```diff
@@ -0,0 +1,529 @@
+            return this;
+        }
+
+        /**
+         * Defines the limit of errors per seconds after which a host will be considered INVALID
+         * for the {@link ErrorAwarePolicy}.
+         * <p/>
+         * The error tracker is going to get notified right away, by the {@link Cluster} it is registered to, of each error
+         * happening for each hosts. In addition to that the error tracker will get updated on a fixed period basis.
+         * During each of these updates, the {@link PerHostErrorTracker} can calculate for each host (thanks to the notifications
+         * from the cluster) a rate of errors per seconds based on the update frequency and the number of errors
+         * since the last update. The calculation will be done as such:
+         * <p/>
+         * Given :
+         * <ul>
+         *  <li>the number of errors happened since the last update \(nbErrors\),</li>
+         *  <li>the update frequency in milliseconds \(updatePeriod\) (see {@link #withUpdatePeriodMillis})</li>
+         *  <li>the rate 'milliseconds per seconds' \(millisToSec\) (= 1000)</li>
+         * </ul>
+         * <p/>
+         * The rate for each host \(errorsPerSecHostRate\) will be:
+         * \[
+         * errorsPerSecHostRate = \frac{nbErrors * updatePeriod}{millisToSec}
+         * \]
+         * <p/>
+         * This will give to the tracker an estimation of the number of errors/seconds currently happening for a host.
+         * <p/>
+         * The threshold given in argument to this method will also be converted the same way into a rate
+         * (\(thresholdRate\)) in order to be compared with the \(errorsPerSerHostRate\).
+         * <p/>
+         * If the \(errorsPerSerHostRate\) estimation is <b>higher than or equal to</b> the \(thresholdRate\), the host
+         * will be ignored, {@link PerHostErrorTracker#isHostValid(Host)} will return {@code false}. Otherwise,
+         * the host will still be considered valid.
+         * <p/>
+         * Default value for the threshold is 10 errors/second.
+         *
+         * @param errorsPerSecThreshold the errors per seconds inclusion threshold, without taking into account the
+         *                              update frequency (updatePeriod).
+         * @return this {@link Builder} instance, for method chaining.
+         */
+        public Builder withInclusionThreshold(int errorsPerSecThreshold) {
+            this.errorsPerSecThreshold = errorsPerSecThreshold;
+            return this;
+        }
+
+        /**
+         * Plug in a custom {@link ErrorAwarePolicyTracker} to use in the {@link ErrorAwarePolicy}. See the
+         * {@link ErrorAwarePolicyTracker} for more information.
+         *
+         * @param errorTracker the tracker to plug in.
+         * @return this {@link Builder} instance, for method chaining.
+         */
+        public Builder withErrorTracker(ErrorAwarePolicyTracker errorTracker) {
```

Comment:
Agree with that, I will write a builder for the PerHostTracker that take only the options specific to it. However on the flip side it adds more boiler plate code for users that don't provide a ErrorTracker but only want to configure the threshold for ex. I.e.
```
policy = ErrorAwarePolicy.builder().withUpdatePeriod(100).withErrorTracker(PerHostErrorTracker.builder().withThreshold(2).build()).build();
```
VS
```
policy = ErrorAwarePolicy.builder().withUpdatePeriod(100).withThreshold(2).build();
```